### ISSUE TITLE
cannot strip expression in this position - outer attributes not allowed

### ISSUE THREAD

**Issue by dkm** [Rust-GCC/gccrs#391]
Not really sure what is the meaning of the message for : 
```
fn foo (e: &str) -> &str {
    &""
}
```
```
slice1.rs:2:6: error: cannot strip expression in this position - outer attributes not allowed
```

FTR, rustc gives : 
```
warning: unused variable: `e`
 --> .../git/gccrs/gcc/testsuite/rust.test/unsupported/slice1.rs:1:9
  |
1 | fn foo (e: &str) -> &str {
  |         ^ help: if this is intentional, prefix it with an underscore: `_e`
  |
  = note: `#[warn(unused_variables)]` on by default

warning: function is never used: `foo`
 --> .../git/gccrs/gcc/testsuite/rust.test/unsupported/slice1.rs:1:4
  |
1 | fn foo (e: &str) -> &str {
  |    ^^^
  |
  = note: `#[warn(dead_code)]` on by default

warning: 2 warnings emitted
```

**Comment by SimplyTheOther** [comment]
I believe the error results from this:

The string literal given is empty. This is incidentally also the representation of an “invalid literal”, as defined by the is_error method in Literal. This was a typo - it was meant to be an empty char only that is invalid (since you can’t have an empty char).

The LiteralExpr that wraps this Literal into an expression is also marked as “invalid” as a result. 

In macro expansion, all expressions (among other things) are recursively checked for being invalid - being invalid is a marker to “strip” them. You obviously can’t strip components of operator expressions as they no longer make sense without operands, and that is why this error occurs - the macro expander finds that the LiteralExpr used as the operand for the BorrowExpr is invalid, but knows it can’t strip it, so it gives this error message.

TL;DR: We need to add a condition stating that the “type” of the literal must also be char to be invalid (as well as being empty).

**Comment by dkm** [comment]
Thanks for the details! I think I'll have to read a bit more to really understand what you are describing :D
